src/ring_theory/nilpotent.lean

```diff
@@ -0,0 +1,107 @@
+/-
+Copyright (c) 2021 Oliver Nash. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Oliver Nash
+-/
+import data.nat.choose.sum
+
+/-!
+# Nilpotent elements
+
+## Main definitions
+
+  * `is_nilpotent`
+  * `is_nilpotent_neg_iff`
+  * `commute.is_nilpotent_add`
+  * `commute.is_nilpotent_mul_left`
+  * `commute.is_nilpotent_mul_right`
+  * `commute.is_nilpotent_sub`
+
+-/
+
+universes u
+
+variables {R : Type u} {x y : R}
+
+/-- An element is said to be nilpotent if some natural-number-power of it equals zero.
+
+Note that we require only the bare minimum assumptions for the definition to make sense. Even
+`monoid_with_zero` is too strong since nilpotency is important in the study of rings that are only
+power-associative. -/
+def is_nilpotent [has_zero R] [has_pow R ℕ] (x : R) : Prop := ∃ (n : ℕ), x^n = 0
+
+lemma monoid_with_zero.pow_eq_zero_of_le
```

Comment:
Should this move to another file?

Reply:
Right of course. I think I've found a sensible home for it in `algebra/group_power/basic.lean`.